src/Core/Environment.php

```diff
@@ -151,4 +166,92 @@ public static function getTimeLimitMax()
     {
         return static::$timeLimitMax;
     }
+
+    /**
+     * Get value of environment variable
+     *
+     * @param string $name
+     * @return mixed Value of the environment variable, or null if not set
+     */
+    public static function getenv($name)
+    {
+        switch (true) {
+            case array_key_exists($name, static::$env):
+                return static::$env[$name];
+            case array_key_exists($name, $_ENV):
+                return $_ENV[$name];
+            case array_key_exists($name, $_SERVER):
+                return $_SERVER[$name];
+            default:
+                return getenv($name);
+        }
+    }
+
+    /**
+     * Set environment variable using php.ini syntax.
+     * Acts as a process-isolated version of putenv()
+     * Note: This will be parsed via parse_ini_string() which handles quoted values
+     *
+     * @param string $string Setting to assign in KEY=VALUE or KEY="VALUE" syntax
+     */
+    public static function putenv($string)
+    {
+        // Parse name-value pairs
+        $envVars = parse_ini_string($string) ?: [];
+        foreach ($envVars as $name => $value) {
+            self::setEnv($name, $value);
+        }
+    }
+
+    /**
+     * Load environment variables from .env file
+     *
+     * @param string $path Path to the file
+     * @return array List of values parsed as an associative array
+     */
+    public static function putenvFromFile($path)
+    {
+        // Parse and cleanup content
+        $contents = implode(
+            PHP_EOL,
+            array_filter(array_map(
+                function ($line) {
+                    // Strip all # comments, including start or end of string, but not in quoted strings
+                    // '#comment', 'var=thing#comment', or 'var="not#comment"#comment
+                    return preg_replace('/((^[^"]*)|(^[^"]+("[^"]*){2}))#.*$/', '$1', $line);
+                },
+                explode(PHP_EOL, file_get_contents($path))
+            ))
+        );
+
+        // Pass to parse_ini_string
+        $variables = parse_ini_string($contents) ?: [];
+        foreach ($variables as $name => $value) {
+            // Don't overload vars
+            if (static::getenv($name) === false) {
```

Comment:
could we have overloading vars as a second param for this method?

Reply:
Yes that's a good idea thank you. :)